main/accounts/index.js

```diff
@@ -565,17 +565,98 @@ class Accounts extends EventEmitter {
     delete this.accounts[address]
   }
 
-  setGasPrice (price, handlerId, cb) {
-    if (!price || isNaN(parseInt(price, 'hex')) || parseInt(price, 'hex') < 0) return cb(new Error('Invalid price'))
+  invalidValue (fee) {
+    return (!fee || isNaN(parseInt(fee, 'hex')) || parseInt(fee, 'hex') < 0)
+  }
+
+  setBaseFee (baseFee, handlerId, cb) {
+    if (this.invalidValue(baseFee)) return cb(new Error('Invalid base fee'))
+    if (!this.current()) return
+    if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
+      if (parseInt(baseFee, 'hex') > 9999 * 1e9) baseFee = '0x' + (9999 * 1e9).toString(16)
+
+      const priorityFee  = this.current().requests[handlerId].data.maxPriorityFeePerGas
+      const gasLimit = this.current().requests[handlerId].data.gasLimit
+
+      const limit = parseInt(gasLimit, 'hex')
+
+      let fee = parseInt(baseFee, 'hex') + parseInt(priorityFee, 'hex')
+      if (fee * limit > FEE_MAX) {
+        log.warn('Operation would set fee over hard limit')
+        fee = '0x' + (Math.floor(FEE_MAX / limit)).toString(16)
+      } else {
+        fee = '0x' + fee.toString(16)
+      }
+      
+      this.current().requests[handlerId].data.maxFeePerGas = fee
+      this.current().update()
+      cb()
+    }
+  }
+
+  setPriorityFee (priorityFee, handlerId, cb) {
+    if (this.invalidValue(priorityFee)) return cb(new Error('Invalid priority fee'))
+    if (!this.current()) return
+    if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
+      if (parseInt(priorityFee, 'hex') > 9999 * 1e9) priorityFee = '0x' + (9999 * 1e9).toString(16)
+      
+      const maxFeePerGas = parseInt(this.current().requests[handlerId].data.maxFeePerGas, 'hex')
+      const gasLimit = parseInt(this.current().requests[handlerId].data.gasLimit, 'hex')
+      const maxPriorityFeePerGas = parseInt(this.current().requests[handlerId].data.maxPriorityFeePerGas, 'hex')
+      const baseFee = maxFeePerGas - maxPriorityFeePerGas
```

Comment:
do we need to back out the base fee? can we just add the diff of the old priority fee and the new one?
`newMaxFeePerGas = maxFeePerGas + (newMaxPriorityFeePerGas - maxPriorityFeePerGas)`

Reply:
It's useful for the fee limiter logic